core/src/main/java/com/scalar/db/transaction/consensuscommit/UncommittedRecordException.java

```diff
@@ -1,38 +1,38 @@
 package com.scalar.db.transaction.consensuscommit;
 
 import com.google.common.collect.ImmutableList;
+import com.scalar.db.api.Selection;
 import com.scalar.db.exception.transaction.CrudConflictException;
-import java.util.ArrayList;
-import java.util.Collections;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import java.util.List;
 
 public class UncommittedRecordException extends CrudConflictException {
-  private final List<TransactionResult> results;
 
-  public UncommittedRecordException(
-      TransactionResult result, String message, String transactionId) {
-    this(result, message, null, transactionId);
-  }
+  private final Selection selection;
+  private final List<TransactionResult> results;
 
+  @SuppressFBWarnings("EI_EXPOSE_REP2")
   public UncommittedRecordException(
-      TransactionResult result, String message, Throwable cause, String transactionId) {
-    super(message, cause, transactionId);
-    results = Collections.singletonList(result);
+      Selection selection, TransactionResult result, String message, String transactionId) {
+    super(message, transactionId);
+    this.selection = selection;
+    results = ImmutableList.of(result);
   }
 
+  @SuppressFBWarnings("EI_EXPOSE_REP2")
   public UncommittedRecordException(
```

Comment:
This seems not used. Do you want to keep it for future use?

Reply:
I left this here because it was originally here. But yes, we don't need to leave it. I'll remove it. Thanks.

Reply:
Fixed in 5c4d6a7120eb109a81a6a3c7ca8fb897d481e05b.